### PR TITLE
Remove dead code from System.Json

### PR DESCRIPTION
**Pull request by hughbe**
And increase CC by hitting non C# (IL) branches

/cc @stephentoub

### CODE REVIEW

File: src/System.Json/src/System/Json/JsonValue.cs [hunk 1]
```diff
@@ -70,26 +71,14 @@ private static JsonValue ToJsonValue(object ret)
             }
 
             if (ret is bool) return new JsonPrimitive((bool)ret);
-            if (ret is byte) return new JsonPrimitive((byte)ret);
```
Comment (stephentoub) [inline src/System.Json/src/System/Json/JsonValue.cs]: Can you outline for me how we know these are dead?

**Comment by stephentoub** [comment]
@dotnet-bot test this please (PR submitted when there was a build break)

File: src/System.Json/src/System/Json/JsonValue.cs (hunk 1 above)
Comment (hughbe) [inline src/System.Json/src/System/Json/JsonValue.cs]: Sure -
Basically, System.Json uses an internal `JavaScript` reader to parse data. Firstly, it is internal so not extensible.
The parsing functionality is found here: https://github.com/hughbe/corefx/blob/09535912c6619a51e14b85a63828e565677add6f/src/System.Json/src/System/Json/JavaScriptReader.cs#L39

Basically, the method can only return the following:
- a List<object>: [here](https://github.com/hughbe/corefx/blob/09535912c6619a51e14b85a63828e565677add6f/src/System.Json/src/System/Json/JavaScriptReader.cs#L57)
- object[]: [here](https://github.com/hughbe/corefx/blob/09535912c6619a51e14b85a63828e565677add6f/src/System.Json/src/System/Json/JavaScriptReader.cs#L76)
- Dictionary<string, object>: [here](https://github.com/hughbe/corefx/blob/09535912c6619a51e14b85a63828e565677add6f/src/System.Json/src/System/Json/JavaScriptReader.cs#L85)
- object[]: [here](https://github.com/hughbe/corefx/blob/09535912c6619a51e14b85a63828e565677add6f/src/System.Json/src/System/Json/JavaScriptReader.cs#L112)
- bool: [here](https://github.com/hughbe/corefx/blob/09535912c6619a51e14b85a63828e565677add6f/src/System.Json/src/System/Json/JavaScriptReader.cs#L116) or [here](https://github.com/hughbe/corefx/blob/09535912c6619a51e14b85a63828e565677add6f/src/System.Json/src/System/Json/JavaScriptReader.cs#L120)
- string: [here](https://github.com/hughbe/corefx/blob/09535912c6619a51e14b85a63828e565677add6f/src/System.Json/src/System/Json/JavaScriptReader.cs#L127)
- int: [here](https://github.com/hughbe/corefx/blob/09535912c6619a51e14b85a63828e565677add6f/src/System.Json/src/System/Json/JavaScriptReader.cs#L260)
- ulong: [here](https://github.com/hughbe/corefx/blob/09535912c6619a51e14b85a63828e565677add6f/src/System.Json/src/System/Json/JavaScriptReader.cs#L272)
- decimal: [here](https://github.com/hughbe/corefx/blob/09535912c6619a51e14b85a63828e565677add6f/src/System.Json/src/System/Json/JavaScriptReader.cs#L279)
- double: [here](https://github.com/hughbe/corefx/blob/09535912c6619a51e14b85a63828e565677add6f/src/System.Json/src/System/Json/JavaScriptReader.cs#L318)

Comment (stephentoub) [inline src/System.Json/src/System/Json/JsonValue.cs]: I see. Ok, thanks.

**Pull request merged**